lib/styles/components/_buttons.scss

```diff
@@ -3,48 +3,183 @@
 @import '../settings/spacings';
 @import '../settings/radiuses';
 @import '../settings/typography';
+@import '../settings/buttons';
 
-$default-lighter-shadow: 0 4px 12px rgba($color-firefly-black, 0.35);
+$default-lighter-shadow: 0 4px 12px rgba($color-firefly-black-rgb, 0.35);
 
-@mixin outlinedColor($color) {
-	color: rgba($color, 1);
+@mixin setButtonColor($color: '', $color-hovered: '', $color-focused: '', $color-disabled: '') {
+	@if $color != '' {
+		color: $color;
+	}
 
-	&:disabled,
-	&.-disabled {
-		color: rgba($color, 1 - $button-disabled-alpha);
+	@if $color-hovered != '' {
+		&:hover,
+		&.-hovered,
+		&:active,
+		&.-active {
+			color: $color-hovered;
+		}
+	}
+
+	@if $color-focused != '' {
+		&:focus,
+		&.-focused {
+			color: $color-focused;
+		}
+	}
+
+	@if $color-disabled != '' {
+		&.-disabled {
+			color: $color-disabled;
+		}
 	}
 }
 
-@mixin textColor($color, $hover-color: '', $focus-color: '', $disabled-color: '') {
-	@if $hover-color == '' {
-		$hover-color: mix($color, $color-firefly-black, (1 - $button-hover-alpha) * 100%);
+@mixin setButtonBackground(
+	$background: '',
```

Comment:
making all fields optional doesn't tell devs which are required and which can be ommited. Now, if I ommit wrong fields, we'll end up with bug. Do you think all should be optional?

Reply:
Every usage of this mixin have all parameters, so I'm removing optional values.